hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/AbstractRealtimeRecordReader.java

```diff
@@ -209,6 +209,11 @@ public static Schema generateProjectionSchema(Schema writeSchema, List<String> f
     return projectedSchema;
   }
 
+  public static Map<String, Field> getNameToFieldMap(Schema schema) {
+    return schema.getFields().stream().map(r -> Pair.of(r.name().toLowerCase(), r))
```

Comment:
do we need `toLowerCase()`. avro is case sensitive?

Reply:
Yep looks like Avro is case sensitive. https://avro.apache.org/docs/1.8.2/spec.html#names

Possibly thats also the reason why this code was doing `lowercase` conversion in the first place: 

https://github.com/apache/incubator-hudi/blob/master/hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/AbstractRealtimeRecordReader.java#L195

Reply:
I understand this is not related to the change here. can you just add a todo to consider case sensitivity down the line or may be file a JIRA. May be we need to understand what happens when fields just differ in case

Reply:
Created a Jira https://issues.apache.org/jira/browse/HUDI-303 where we can follow up on.